### PR TITLE
Fix wobbling tests when specialize_float=False

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #141236
* __->__ #140893
* #138922
* #138915

At a high level we plan on supporting dynamic float arguments (like learning rate) by tensorifying float computation. This PR is whack-a-moling a bunch of test failures when I turn the specialize_float=False flag. You can read more about the project and motivation in this gdoc: https://docs.google.com/document/d/1HswUSp9H6mg8Vg27mhRk8YzC9q_uf63b6wz-gwx65BQ/edit?pli=1&tab=t.0#heading=h.xvyiqp8tuje6

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 1]
```diff
@@ -241,6 +241,10 @@ def _test_code_common(
 
 
 class TestPatternMatcher(TestPatternMatcherBase):
+    # The pattern match for this is kind of broken. It'll match if negative_slope
+    # is either a float or a tensor. I'll cc the person who wrote this test/match
+    # on the diff to see if they can help me fix it.
+    @torch._dynamo.config.patch(specialize_float=True)
```
Comment (bobrenjc93) [inline test/inductor/test_mkldnn_pattern_matcher.py:247]: @yanbing-j @zhuhaozhe any chance I can get your help taking a look on this? Basically if you check out this PR and remove this line, you'll see the pattern match breaking.

**Comment by bobrenjc93** [comment]
@dlibenzi  @asuhan any chance I can get your help with the XLA test failures? I have no idea why they are failing and how to fix them.

File: test/inductor/test_compiled_optimizers.py [hunk 2]
```diff
@@ -175,6 +175,48 @@ class KernelCounts(NamedTuple):
...
+    "test_nadam_cpu": 3,
+    "test_nadam_momentum_decay_cpu": 3,
+    "test_nadam_weight_decay_cpu": 5,
+    "test_nadam_weight_decay_maximize_cpu": 5,
+    "test_nadam_weight_decay_momentum_decay_cpu": 5,
+    "test_nadam_weight_decay_momentum_decay_decoupled_weight_decay_cpu": 5,
+    "test_radam_cpu": 7,
+    "test_radam_eps_cpu": 7,
+    "test_radam_weight_decay_cpu": 7,
+    "test_radam_weight_decay_decoupled_weight_decay_cpu": 7,
```
Comment (bobrenjc93) [inline test/inductor/test_compiled_optimizers.py:219]: I think the following fixes everything

```
KERNEL_COUNT_OVERRIDES = {
    "test_rmsprop_foreach_weight_decay_cpu": 12,
    "test_nadam_foreach_weight_decay_momentum_decay_cpu": 20,
    "test_adamw_amsgrad_capturable_foreach_cuda": 3,
    "test_adamw_amsgrad_capturable_foreach_xpu": 3,
    "test_adamw_amsgrad_capturable_cuda": 6,
    "test_adamw_amsgrad_capturable_xpu": 6,
    "test_adamw_tensor_lr_tensor_betas_amsgrad_capturable_cuda": 6,
    "test_adamw_tensor_lr_tensor_betas_capturable_cuda": 6,
    "test_adamw_tensor_lr_tensor_betas_amsgrad_capturable_xpu": 6,
    "test_adamw_tensor_lr_amsgrad_capturable_cuda": 6,
    "test_adamw_tensor_lr_amsgrad_capturable_xpu": 6,
    "test_adam_tensor_lr_amsgrad_capturable_cuda": 6,
    "test_adam_tensor_lr_amsgrad_capturable_xpu": 6,
    "test_adam_tensor_lr_tensor_betas_amsgrad_capturable_cuda": 6,
    "test_adam_tensor_lr_tensor_betas_capturable_cuda": 6,
    "test_adam_amsgrad_capturable_cuda": 6,
    "test_adam_amsgrad_capturable_xpu": 6,
    "test_adadelta_tensor_lr_capturable_cuda": 6,
    "test_adadelta_tensor_lr_capturable_xpu": 6,
    "test_rmsprop_tensor_lr_capturable_cuda": 6,
    "test_rmsprop_tensor_lr_capturable_xpu": 6,
    "test_adadelta_foreach_weight_decay_maximize_cpu": 12,
    "test_adadelta_foreach_rho_weight_decay_cpu": 12,
    "test_adadelta_foreach_weight_decay_cpu": 12,
    "test_sgd_foreach_momentum_weight_decay_cpu": 16,
    "test_sgd_foreach_momentum_nesterov_weight_decay_cpu": 16,
    "test_sgd_momentum_dampening_foreach_cuda": 5,
    "test_sgd_momentum_dampening_foreach_xpu": 5,
    "test_sgd_momentum_foreach_cuda": 5,
    "test_sgd_momentum_foreach_xpu": 5,
    "test_sgd_weight_decay_maximize_cuda": 4,
    "test_sgd_weight_decay_maximize_xpu": 4,
    "test_sgd_weight_decay_maximize_cpu": 4,
    "test_sgd_weight_decay_cpu": 4,
    "test_sgd_weight_decay_cuda": 4,
    "test_sgd_weight_decay_xpu": 4,
    "test_sgd_momentum_weight_decay_foreach_cuda": 2,
    "test_sgd_momentum_weight_decay_foreach_xpu": 2,
    "test_sgd_momentum_nesterov_weight_decay_foreach_cuda": 2,
    "test_sgd_momentum_nesterov_weight_decay_foreach_xpu": 2,
    "test_sgd_cuda": 4,
    "test_sgd_cpu": 4,
    "test_sgd_xpu": 4,
    "test_adagrad_initial_accumulator_value_weight_decay_foreach_xpu": 2,
    "test_adagrad_lr_decay_weight_decay_foreach_xpu": 2,
    "test_adagrad_weight_decay_foreach_xpu": 2,
    "test_adagrad_weight_decay_maximize_foreach_xpu": 2,
    "test_adagrad_tensor_lr_cpu": 6,
    "test_adagrad_tensor_lr_cuda": 6,
    "test_adagrad_tensor_lr_xpu": 6,
    "test_adamax_tensor_lr_weight_decay_capturable_cuda": 6,
    "test_adamax_tensor_lr_weight_decay_capturable_xpu": 6,
    "test_asgd_tensor_lr_weight_decay_maximize_capturable_cuda": 5,
    "test_asgd_tensor_lr_weight_decay_maximize_capturable_xpu": 8,
    "test_nadam_tensor_lr_weight_decay_momentum_decay_decoupled_weight_decay_capturable_cuda": 6,
    "test_nadam_tensor_lr_weight_decay_momentum_decay_decoupled_weight_decay_capturable_xpu": 9,
    "test_radam_tensor_lr_capturable_weight_decay_decoupled_weight_decay_cuda": 6,
    "test_radam_tensor_lr_capturable_weight_decay_decoupled_weight_decay_xpu": 6,
    "test_sgd_tensor_lr_cpu": 2,
    "test_sgd_tensor_lr_cuda": 2,
    "test_sgd_tensor_lr_xpu": 2,
    "test_rmsprop_weight_decay_cpu": 6,
    "test_rmsprop_weight_decay_centered_cpu": 6,
    "test_radam_weight_decay_maximize_cpu": 7,
    "test_rmsprop_maximize_weight_decay_cpu": 6,
    "test_rmsprop_maximize_cpu": 6,
    "test_rmsprop_cpu": 6,
    "test_adadelta_cpu": 6,
    "test_adadelta_maximize_cpu": 6,
    "test_adadelta_rho_weight_decay_cpu": 6,
    "test_adadelta_weight_decay_cpu": 6,
    "test_adadelta_weight_decay_maximize_cpu": 6,
    "test_adagrad_cpu": 6,
    "test_adagrad_initial_accumulator_value_weight_decay_cpu": 6,
    "test_adagrad_lr_decay_weight_decay_cpu": 6,
    "test_adagrad_weight_decay_cpu": 6,
    "test_adagrad_weight_decay_maximize_cpu": 6,
    "test_adam_cpu": 6,
    "test_adam_weight_decay_cpu": 6,
    "test_adam_weight_decay_maximize_cpu": 6,
    "test_adamax_cpu": 6,
    "test_adamax_maximize_cpu": 6,
    "test_adamax_weight_decay_cpu": 6,
    "test_adamax_weight_decay_maximize_cpu": 6,
    "test_adamw_cpu": 6,
    "test_adamw_weight_decay_cpu": 6,
    "test_adamw_weight_decay_maximize_cpu": 6,
    "test_asgd_cpu": 3,
    "test_asgd_lambd_cpu": 3,
    "test_asgd_maximize_cpu": 3,
    "test_asgd_t0_cpu": 3,
    "test_asgd_weight_decay_cpu": 3,
    "test_asgd_weight_decay_maximize_cpu": 3,
    "test_nadam_cpu": 3,
    "test_nadam_momentum_decay_cpu": 3,
    "test_nadam_weight_decay_cpu": 5,
    "test_nadam_weight_decay_maximize_cpu": 5,
    "test_nadam_weight_decay_momentum_decay_cpu": 5,
    "test_nadam_weight_decay_momentum_decay_decoupled_weight_decay_cpu": 5,
    "test_radam_cpu": 7,
    "test_radam_eps_cpu": 7,
    "test_radam_weight_decay_cpu": 7,
    "test_radam_weight_decay_decoupled_weight_decay_cpu": 7,
    "test_adagrad_cuda": 6,
    "test_adagrad_initial_accumulator_value_weight_decay_cuda": 6,
    "test_adagrad_lr_decay_weight_decay_cuda": 6,
    "test_adagrad_weight_decay_cuda": 6,
    "test_adagrad_weight_decay_maximize_cuda": 6,
    "test_asgd_recompile_single": 16,
}
```

**Comment by huydhn** [comment]
Running @pytorchbot drci will refresh the comment, but let me check why it wasn't updated in the first place

**Comment by huydhn** [comment]
Just FYI, the issue with the stale list of failures on the PR has been fixed by https://github.com/pytorch/test-infra/pull/5963
